lib/jnpr/junos/utils/sw.py

```diff
@@ -430,11 +494,20 @@ def _progress(report):
                 _progress(
                     "validating software against current config,"
                     " please be patient ...")
-                v_ok = self.validate(remote_package, dev_timeout=timeout)
-                if v_ok is not True:
-                    return v_ok  # will be the string of output
+                v_ok = self.validate(remote_package, issu, dev_timeout=timeout)
 
-            if self._multi_RE is False:
+                if v_ok is not True:
+                    return v_ok
+
+            if issu is True:
```

Comment:
ISSU and NSSU only make sense when `self._multi_RE is True`. Currently the code tries them even if `self._multi_RE is False`. Maybe this is OK because the RPCs themselves will likely fail in this case. However, I'm wondering if it makes more sense for us to only attempt ISSU/NSSU when `self._multi_RE is True`.